### ISSUE TITLE
RA.contained/RA.notContained

### ISSUE THREAD

**Issue by guillaumearm** [char0n/ramda-adjunct#392]
Also known as `contained` or `within`, I prefer `oneOf` anyway.

```js
const oneOf = R.flip(R.contains)
```

Ref #38

**Comment by Undistraction** [comment]
I like it, but don't understand why the name `oneOf`.

**Comment by guillaumearm** [comment]
For me it's very more clear.

e.g.

```js
R.reject(RA.oneOf([1, 2, 3, 4]));
```

```js
if (RA.oneOf([1, 2, 3, 4]), n) {
  console.log('n is one of them');
}
```

Sometimes, we had to choose semantic over consistance, what do you think ?

Anyway, `within`, `isContained`, `isContainedBy` and `containedBy` are fine too.

**Comment by char0n** [comment]
I vote for `within`, `isWithin`, or fallback to `containesFlipped`

`isWithin` evokes predicate and `containesFlipped` is a last resort. So for me `within` is a clear winner.


`oneOf` evokes a functiona that works similiar to [lodash.sample](https://lodash.com/docs/4.17.5#sample)

**Comment by guillaumearm** [comment]
I don't like `containsFlipped`, I prefer use `flip(contains)` directly.

But `within` or `isWithin` is fine for me ;-)

**Comment by char0n** [comment]
Ok votes pls

`+1 for within`

**Comment by Undistraction** [comment]
I agree that `containsFlipped` is the worst option, but prefer `isContained` or `contained` to `isWithin` or `within` because of the link to `contains` and the fact it is a clear opposite.

`+1 for contained`

**Comment by guillaumearm** [comment]
@Undistraction : you coninced me.

`+1 for contained`

**Comment by char0n** [comment]
it's `contained` then

**Comment by GingerPlusPlus** [comment]
I'd alias it as `within` tho, dunno, it just sounds better to me in cases like `RA.within([1, 2, 3, 4])`.

Or to `oneOf`. Even if some of you don't get it, I do. It's not about array includes, it's more of `value === 1 || value === 2 || value === 3 || value === 4`.

**Comment by guillaumearm** [comment]
Yup, that's why I prefer `oneOf`.
For me it's more like a `anyPass` but with values.

**Comment by Undistraction** [comment]
@gingerplusplus It`s not about not getting it. It's about not agreeing.

I just don't agree. It doesn't expose any predicate. The name `oneOf` makes more sense as an alias for `anyOf` or an xor version of `anyOf`. It might make sense if it was `oneOfEquals` but I prefer `contained` or `within`.

**Comment by char0n** [comment]
> I'd alias it as within tho, dunno, it just sounds better to me in cases like RA.within([1, 2, 3, 4]).
>
> Or to oneOf. Even if some of you don't get it, I do. It's not about array includes, it's more of value === 1 > || value === 2 || value === 3 || value === 4.

Please avoid passing judgments here. There are various people here with various opinions, and that's what is pushing RA lib forward. Some time there is an agreement, some time there is not and we vote.

Please refer to our [CODE OF CONDUCT](https://github.com/char0n/ramda-adjunct/blob/master/CODE_OF_CONDUCT.md) which set the standards for creating positive and welcoming environment.

**Comment by guillaumearm** [comment]
I would like to make a PR for this,
can we decide for a name ?

If we vote, I vote `RA.contained`.

**Comment by Undistraction** [comment]
> it's contained then

We already voted and `contained` won.

**Comment by GingerPlusPlus** [comment]
I'd love to have an alias... but whatever, it can easily be added later, removing is harder.

So, 100% fine with `contained` only.

**Comment by guillaumearm** [comment]
@Undistraction : Oups sorry for that.

@GingerPlusPlus, you can alias it yourself : 
```js
import { contained as oneOf } from 'ramda-adjunct';
// or
import { contained as within } from 'ramda-adjunct';
```

**Comment by Undistraction** [comment]
@guillaumearm No need to apologise.

**Comment by char0n** [comment]
Yes, let's not alias this to keep `oneOf` and `within` free for feature usage.
For completness, commonjs aliasing
```js
const { contained: oneOf } = require('ramda-adjunct');

**Comment by Undistraction** [comment]
@guillaumearm Added this to 2.7.0. No pressure.

**Comment by tommmyy** [comment]
https://github.com/tommmyy/ramda-extension/blob/master/packages/ramda-extension/src/includes.js

https://github.com/tommmyy/ramda-extension/blob/master/packages/ramda-extension/src/notInclude.js

**Comment by Undistraction** [comment]
@tommmyy Thanks for the links. @guillaumearm Fancy adding a `notContained`?

**Comment by guillaumearm** [comment]
@Undistraction : sure

**Comment by Undistraction** [comment]
@guillaumearm I would put them both in a single PR

**Comment by char0n** [comment]
This has been implemented in v2.8.0 already. THank you!
